csharp/ql/src/semmle/code/csharp/controlflow/internal/Splitting.qll

```diff
@@ -582,11 +645,11 @@ module FinallySplitting {
       succ = any(FinallyControlFlowElement fcfe |
           if fcfe.isEntryNode()
           then
-            // Entering a nested `finally` block normally must remember the outer split
-            c instanceof NormalCompletion
+            // entering a nested `finally` block
+            nestLevel(fcfe.getTryStmt()) > this.getNestLevel()
```

Comment:
No, we need to remember the outer splitting, even if entering the nested finally block with a non-normal completion. Example:
```
    void M()
    {
        try
        {
            throw new ExceptionA();
        }
        finally
        {
            try
            {
                try
                {
                    throw new ExceptionB();
                }
                finally
                {
                }
            }
            catch
            {
            }
        }
    }
```
We always enter the inner `finally` block with an `ExceptionB` completion, but because the inner finally block is wrapped in a `try` that ignores the exception, we need to still remember the `ExceptionA` completion for the outer `finally` block, in order for the body to end with an `ExceptionA` completion.